app/lib/features/news/pages/add_news_page.dart

```diff
@@ -405,6 +374,17 @@ class AddNewsState extends ConsumerState<AddNewsPage> {
               slidePost.backgroundColor?.value,
             ),
           );
+
```

Comment:
While you check in the UI that one can't select an event before having selected a space, there is still the unlikely case that the user changed the space after selecting the event, thus leading to a faulty reference.

We could check for that here, like have a `mustBeInSpaceId = ` on the `newsReferenceModel` and ensure that they match up and refuse to submit, or [when selecting a space](https://github.com/acterglobal/a3/pull/1450/files#diff-965bf3e8329d6f06301669ae25a78c85bc589f2d7846e778d7c34003305a9f6fR31-R36) we could clear the event to ensure this never happens ...

We need to do at least one of the two to prevent this accidental food gun to go off.

Reply:
Let's clear event select upon Space Selection changes so that this issue may not exists.

Reference:
https://github.com/acterglobal/a3/assets/51526480/95dabfa6-ed21-4c40-af79-eea1d9a86dbd